blog-posts/the-modern-code-reviewer-2021.md

```diff
@@ -0,0 +1,219 @@
+
+Other folks exclude themselves from the **common review** because they feel intimidated by their lack of knowledge, the complexity of the code changes or the superior level of the reviewee. The **shallow review** is an excellent instrument to encourage those people in participating in the daily code review routines and start gaining knowledge and experience in that important field. But the shallow review is also a great tool for seasoned developers. You can pick it and quickly confirm a change with the expectation that you did not dug as deep as usually.
+
+Finally the **expert review** is your instrument to ensure an **in-depth review** of content that is pioneering a new approach or introducing security relevant changes.
+
+The moment you understand that you have a choice when you are requested for a code review, you can distribute your time more evenly and steer away from situations that would cause you to be the bottleneck. It's also great to clarify expectations as you can proclaim the way to reviewed. An "I only did a shallow review" makes clear that the PR will deliver a working change without everything being perfect in terms of code quality, architecture and more.
+
+Being team, the knowledge of different code review depths helps you to prevent large backlogs of PRs and frustrated reviewees waiting for some feedback. You have a way to motivate people to take part in the review process who previously lacked the necessary confidence.
+
+# Further improvements
+
+We focused exclusively on the reviewer yet. Being a reviewee, you can enhance the chances that one of the review types will be seen feasible by the reviewer if you make your PR more accessible. Provide a clear description. Repeat parts of the Jira ticket, make the PR small enough, if it's big tell the reviewer why and tell where to focus on. Give clear instructions to run the code and to follow your changes.
+
+This will greatly improve your chances to receive a review as you lower the effort to quickly check into your PR and you will encourage people who lack confidence or knowledge to provide their review too.
+
+I'm planning to provide a followup to this blog post describing the ideal pull request which will focus on the reviewer's duty.
+
+# Conclusion
+
+If you want to use the described approach, you need a working code review environment where people trust each other. Shallow reviews won't magically solve your toxic environment. You have to fix the foundation before opimizing your PR throughput.
+
+Encourage the less experienced colleagues to start with shallow code reviews to actively take part in shaping the future of your code base. Pick your own code review type wisely though. If everyone gets lazy and provides only shallow reviews, your overall coding quality might suffer, because nobody takes their time to check the important details of a code change.
```

Comment:
```suggestion
Encourage the less experienced colleagues to start with shallow code reviews to actively take part in shaping the future of your code base. Pick your own code review type wisely though. If everyone gets lazy and provides only shallow reviews, your overall code quality might suffer, because nobody takes their time to check the important details of a code change.
```